tests/workflow/test_transforms_execution.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit and integration tests for execution of transform programs."""
+# pylint: disable=not-callable
+from typing import Sequence, Callable
+from functools import partial
+import copy
+import numpy as np
+
+import pennylane as qml
+
+dev = qml.device("default.qubit", wires=2)
+
+
+@qml.qnode(device=dev)
+def qnode_circuit(a):
+    """QNode circuit."""
+    qml.CNOT(wires=[0, 1])
+    qml.PauliX(wires=1)
+    qml.RX(a, wires=0)
+    return qml.expval(qml.PauliZ(wires=0))
+
+
+@qml.transforms.transform
+def shift_transform(
+    tape: qml.tape.QuantumTape, alpha: float
+) -> (Sequence[qml.tape.QuantumTape], Callable):
+    """A valid (dummy) transform that shift all angles."""
+    tape1 = copy.deepcopy(tape)
+    parameters = tape1.get_parameters(trainable_only=False)
+    parameters = [param + alpha for param in parameters]
+    tape1.set_parameters(parameters, trainable_only=False)
```

Comment:
`set_parameters` is getting deprecated. Use `bind_new_parameters` instead. Or if #4345 gets merged before this, use `tape.bind_new_parameters`.

Reply:
#4345 is merged